src/backends/onnxruntime.c

```diff
@@ -255,99 +280,95 @@ RAI_Tensor *RAI_TensorCreateFromOrtValue(OrtValue *v, size_t batch_offset, long
-        const size_t batch_bytesize = sample_bytesize * batch_size;
-
-        char *data = RedisModule_Calloc(batch_bytesize, sizeof(*data));
-        memcpy(data, ort_data + batch_offset * sample_bytesize, batch_bytesize);
-#endif
-
-        ort->ReleaseTensorTypeAndShapeInfo(info);
-
-        // TODO: use manager_ctx to ensure ORT tensor doesn't get deallocated
-        // This applies to outputs
 
-        ret->tensor = (DLManagedTensor){.dl_tensor = (DLTensor){.device = device,
-#ifdef RAI_COPY_RUN_OUTPUT
-                                                                .data = data,
-#else
-#error zero-copy passing output memory from ORT not currently supported
-#endif
-                                                                .ndim = ndims,
-                                                                .dtype = dtype,
-                                                                .shape = shape,
-                                                                .strides = strides,
-                                                                .byte_offset = 0},
-                                        .manager_ctx = NULL,
-                                        .deleter = NULL};
-
-        return ret;
+        size_t total_byte_size = elem_count * data_type_size;
+        size_t sample_byte_size = total_byte_size / total_batch_size;
+        memcpy(RAI_TensorData(output_tensor), ort_data + batch_offset * sample_byte_size,
+               RAI_TensorByteSize(output_tensor));
+    } else {
+        // Calculate the blob size for this tensor and allocate space for it
+        size_t element_index = batch_offset * (output_tensor_len / batch_size);
+        size_t blob_len = 0;
+        for (size_t i = 0; i < output_tensor_len; i++) {
+            size_t str_element_len;
+            ONNX_VALIDATE_STATUS(
+                ort->GetStringTensorElementLength(v, element_index++, &str_element_len));
+            blob_len +=
+                str_element_len + 1; // Add space for null character at the end of every string
+        }
+        output_tensor->blobSize = blob_len;
+        output_tensor->tensor.dl_tensor.data = RedisModule_Alloc(blob_len);
+        char *tensor_blob = RAI_TensorData(output_tensor);
+
+        // Go over again and set tensor data elements one by one
+        element_index = batch_offset * (output_tensor_len / batch_size);
+        uint64_t *offsets = RAI_TensorStringElementsOffsets(output_tensor);
+        offsets[0] = 0;
+        for (size_t i = 0; i < output_tensor_len; i++) {
+            size_t str_element_len;
+            ONNX_VALIDATE_STATUS(
+                ort->GetStringTensorElementLength(v, element_index, &str_element_len));
+            ONNX_VALIDATE_STATUS(ort->GetStringTensorElement(v, str_element_len, element_index++,
+                                                             tensor_blob + offsets[i]));
+            *(tensor_blob + offsets[i] + str_element_len) = '\0';
```

Comment:
must?

Reply:
If we want to keep the convention that our blob is a sequence of null-terminated strings, than yes, since ONNX returns the string element without the trailing null-character.